$OpenBSD: patch-panama_h,v 1.1 2011/06/24 07:55:18 dcoppa Exp $

Fix for Panama cipher validation failure (upstream rev 526).
Bug report is about armel, but it has been proven to also prevent
a sigbus crash on sparc64.

--- panama.h.orig	Fri Aug  6 18:44:34 2010
+++ panama.h	Wed Jun  8 16:42:01 2011
@@ -12,7 +12,7 @@ class CRYPTOPP_NO_VTABLE Panama
 {
 public:
 	void Reset();
-	void Iterate(size_t count, const word32 *p=NULL, word32 *z=NULL, const word32 *y=NULL);
+	void Iterate(size_t count, const word32 *p=NULL, byte *output=NULL, const byte *input=NULL, KeystreamOperation operation=WRITE_KEYSTREAM);
 
 protected:
 	typedef word32 Stage[8];
